app/routerTransition.js

```diff
@@ -62,26 +62,30 @@ const willTransitionTo = (nextState, replaceState, callback) => {
     if (!connectionString) connectionString = urls[0].url;
     if (connectionString.indexOf("fake.automatic-selection") !== -1) connectionString = urls[0];
 
-    if (!connectionManager) connectionManager = new Manager({url: connectionString, urls});
-    if (nextState.location.pathname === "/init-error") {
-        return Apis.reset(connectionString, true).init_promise
-        .then(() => {
-            var db;
-            try {
-                db = iDB.init_instance(window.openDatabase ? (shimIndexedDB || indexedDB) : indexedDB).init_promise;
-            } catch(err) {
-                console.log("db init error:", err);
-            }
-            return Promise.all([db, SettingsStore.init()]).then(() => {
-                return callback();
-            }).catch((err) => {
-                console.log("err:", err);
-                return callback();
-            });
+    var onReset = () => {
+        var db;
+        try {
+            db = iDB.init_instance(window.openDatabase ? (shimIndexedDB || indexedDB) : indexedDB).init_promise;
+        } catch(err) {
+            console.log("db init error:", err);
+        }
+        return Promise.all([db, SettingsStore.init()]).then(() => {
+            return callback();
```

Comment:
This reset callback needs to include the promises loading the WalletDb and PrivateKeyActions data, (PrivateKeyActions.loadDbData() and WalletDb.loadDbData()), since the wallet that is loaded depends on the chain id of the websocket API you're connected to. If you switch from the mainnet to the testnet you need to load the correct wallet and private keys. I'm not sure how well those methods play with on the fly switching

Reply:
No problem, can refactor this so the logic is included for both.